src/emulator/auth/operations.ts

```diff
@@ -1011,6 +1041,31 @@ export function setAccountInfoImpl(
   });
 }
 
+function sendOobForEmailReset(state: ProjectState, oldEmail: string, newEmail: string, url: URL) {
+  const mode: string = "recoverEmail";
+
+  // Encode the new email with the OOB code. The old email is received as a URL parameter.
```

Comment:
Ideally both the old state and the new state should be stored in `OobRecord`. Since we are not doing that here, I opted to expose the reset email as a query parameter. We could have set the new email as a query parameter too. While both these options are insecure (which is okay for the auth emulator), I've opted to go with `newEmail` in `OobRecord` and oldEmail` as a parameter to be consistent with other use-cases of `OobRecord`.

Reply:
This needs to be kept on the server side to be aligned with production (and it MUST not be an input parameter on `setAccountInfo`. I think production stores `oldEmail` in `OobRecord` (since the code sent to the `oldEmail`). Why do we need to store `newEmail`?

Reply:
> Why do we need to store newEmail?

Hmm. You're right. I'm using the wrong lookUp method.

Reply:
Done. Added `initialEmail` support.